ext/date/tests/createFromTimestamp.phpt

```diff
@@ -0,0 +1,283 @@
+  ["date"]=>
+  string(26) "1969-12-31 23:59:59.876544"
+  ["timezone_type"]=>
+  int(1)
+  ["timezone"]=>
+  string(6) "+00:00"
+}
+DateTimeImmutable::createFromTimestamp(-0.123456): object(DateTimeImmutable)#%d (3) {
+  ["date"]=>
+  string(26) "1969-12-31 23:59:59.876544"
+  ["timezone_type"]=>
+  int(1)
+  ["timezone"]=>
+  string(6) "+00:00"
+}
+DateTime::createFromTimestamp(0): object(DateTime)#%d (3) {
+  ["date"]=>
+  string(26) "1970-01-01 00:00:00.000000"
+  ["timezone_type"]=>
+  int(1)
+  ["timezone"]=>
+  string(6) "+00:00"
+}
+DateTimeImmutable::createFromTimestamp(0): object(DateTimeImmutable)#%d (3) {
+  ["date"]=>
+  string(26) "1970-01-01 00:00:00.000000"
+  ["timezone_type"]=>
+  int(1)
+  ["timezone"]=>
+  string(6) "+00:00"
+}
+DateTime::createFromTimestamp(0.0): object(DateTime)#%d (3) {
+  ["date"]=>
+  string(26) "1970-01-01 00:00:00.000000"
+  ["timezone_type"]=>
+  int(1)
+  ["timezone"]=>
+  string(6) "+00:00"
+}
+DateTimeImmutable::createFromTimestamp(0.0): object(DateTimeImmutable)#%d (3) {
+  ["date"]=>
+  string(26) "1970-01-01 00:00:00.000000"
+  ["timezone_type"]=>
+  int(1)
+  ["timezone"]=>
+  string(6) "+00:00"
+}
+DateTime::createFromTimestamp(-0.0): object(DateTime)#%d (3) {
+  ["date"]=>
+  string(26) "1970-01-01 00:00:00.000000"
+  ["timezone_type"]=>
+  int(1)
+  ["timezone"]=>
+  string(6) "+00:00"
+}
+DateTimeImmutable::createFromTimestamp(-0.0): object(DateTimeImmutable)#%d (3) {
+  ["date"]=>
+  string(26) "1970-01-01 00:00:00.000000"
+  ["timezone_type"]=>
+  int(1)
+  ["timezone"]=>
+  string(6) "+00:00"
+}
+DateTime::createFromTimestamp(2147483647): object(DateTime)#%d (3) {
+  ["date"]=>
+  string(26) "2038-01-19 03:14:07.000000"
+  ["timezone_type"]=>
+  int(1)
+  ["timezone"]=>
+  string(6) "+00:00"
+}
+DateTimeImmutable::createFromTimestamp(2147483647): object(DateTimeImmutable)#%d (3) {
+  ["date"]=>
+  string(26) "2038-01-19 03:14:07.000000"
+  ["timezone_type"]=>
+  int(1)
+  ["timezone"]=>
+  string(6) "+00:00"
+}
+DateTime::createFromTimestamp(-2147483648): object(DateTime)#%d (3) {
+  ["date"]=>
+  string(26) "1901-12-13 20:45:52.000000"
+  ["timezone_type"]=>
+  int(1)
+  ["timezone"]=>
+  string(6) "+00:00"
+}
+DateTimeImmutable::createFromTimestamp(-2147483648): object(DateTimeImmutable)#%d (3) {
+  ["date"]=>
+  string(26) "1901-12-13 20:45:52.000000"
+  ["timezone_type"]=>
+  int(1)
+  ["timezone"]=>
+  string(6) "+00:00"
+}
+DateTime::createFromTimestamp(-2147483648.5): object(DateTime)#%d (3) {
+  ["date"]=>
+  string(26) "1901-12-13 20:45:51.500000"
+  ["timezone_type"]=>
+  int(1)
+  ["timezone"]=>
+  string(6) "+00:00"
+}
+DateTimeImmutable::createFromTimestamp(-2147483648.5): object(DateTimeImmutable)#%d (3) {
+  ["date"]=>
+  string(26) "1901-12-13 20:45:51.500000"
+  ["timezone_type"]=>
+  int(1)
+  ["timezone"]=>
+  string(6) "+00:00"
+}
+DateTime::createFromTimestamp(%f): DateRangeError: Seconds must be a finite number between %i and %i, %f given
```

Comment:
Why do you say `finite` here, and not `integer`?

Reply:
I used the word "finite" to cover NAN, +/-INF as well and as a FP is allowed as well "integer" would not be fully correct.
If you feel it's confusing it could also be left out -> "... must be a number between ..."